tests/framework/framework.go

```diff
@@ -440,6 +441,67 @@ func (f *Framework) CreatePrometheusServiceInNs(namespace string) (*v1.Service,
 	return f.K8sClient.CoreV1().Services(namespace).Create(service)
 }
 
+// CreateQuotaInNs creates a quota and sets it on the current test namespace.
+func (f *Framework) CreateQuotaInNs(requestCPU, requestMemory, limitsCPU, limitsMemory int64) error {
+	resourceQuota := &v1.ResourceQuota{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-quota",
+			Namespace: f.Namespace.GetName(),
+		},
+		Spec: v1.ResourceQuotaSpec{
+			Hard: v1.ResourceList{
+				v1.ResourceRequestsCPU:    *resource.NewQuantity(requestCPU, resource.DecimalSI),
+				v1.ResourceRequestsMemory: *resource.NewQuantity(requestMemory, resource.DecimalSI),
+				v1.ResourceLimitsCPU:      *resource.NewQuantity(limitsCPU, resource.DecimalSI),
+				v1.ResourceLimitsMemory:   *resource.NewQuantity(limitsMemory, resource.DecimalSI),
+			},
+		},
+	}
+	_, err := f.K8sClient.CoreV1().ResourceQuotas(f.Namespace.GetName()).Create(resourceQuota)
+	if err != nil {
+		ginkgo.Fail("Unable to set resource quota " + err.Error())
+	}
+	return err
+}
+
+// UpdateCdiConfigResourceLimits sets the limits in the CDIConfig object
+func (f *Framework) UpdateCdiConfigResourceLimits(resourceCPU, resourceMemory, limitsCPU, limitsMemory int64) error {
+	config, err := f.CdiClient.CdiV1alpha1().CDIConfigs().Get(common.ConfigName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	config.Spec.PodResourceRequirements = &v1.ResourceRequirements{
+		Requests: map[v1.ResourceName]resource.Quantity{
+			v1.ResourceCPU:    *resource.NewQuantity(resourceCPU, resource.DecimalSI),
+			v1.ResourceMemory: *resource.NewQuantity(resourceMemory, resource.DecimalSI)},
+		Limits: map[v1.ResourceName]resource.Quantity{
+			v1.ResourceCPU:    *resource.NewQuantity(limitsCPU, resource.DecimalSI),
+			v1.ResourceMemory: *resource.NewQuantity(limitsMemory, resource.DecimalSI)},
+	}
+	_, err = f.CdiClient.CdiV1alpha1().CDIConfigs().Update(config)
+	if err != nil {
+		return err
+	}
+	time.Sleep(1 * time.Second)
+	// Make sure the CDIConfig is updated before continuing
```

Comment:
Wow, I really wish the configmap update api was synchronous!

Reply:
This sleep might not actually be enough, it depends on how well the cluster is behaving, but the 1 seconds seems enough in most circumstances. Definitely potential to be flake.

Reply:
So to detail the problem:
Above we update the resource requirements in the CDIConfig object, and below we attempt to verify those changes are actually there. However it appears to be cached because the below will immediately return success, but in reality the CDIConfig object might not have been updated yet. This will cause the resource request/limit to be 0 on the pod, and thus it will successfully launch when it should fail. The timeout is to give the cluster some time to actually propagate the values to the actual object.